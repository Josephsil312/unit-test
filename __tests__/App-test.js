import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import Login from '../App';

it('renders correctly', () => {
  renderer.create(<Login />);
});

test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Login username', () => {
  const { getByText, getByPlaceholderText } = render(
    <Login />,
  );

  fireEvent.changeText(
    getByPlaceholderText('Username')
  );
  fireEvent.press(getByText('Submit'));
});
