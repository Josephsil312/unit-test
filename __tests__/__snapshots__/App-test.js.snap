// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text>
    Login
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Username"
    testID="username"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    testID="password"
    value=""
  />
  <TextInput
    keyboardType="numeric"
    maxLength={10}
    onChangeText={[Function]}
    placeholder="Phone"
    testID="phone"
    value=""
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btnSubmit"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
